File: python-gdb-activities-20/activity17/gdb/command/transaction_command.py
# gdb/command/transaction_command.py
from abc import ABC, abstractmethod
from typing import Optional
from gdb.domain.transaction import Transaction

# ============================================================
# 📝 STEP 1: Define TransactionCommand Interface
#
# INSTRUCTIONS:
#   1. Declare abstract method execute() -> None
#   2. Declare abstract method get_transaction() -> Optional[Transaction]
#
# HINT: Decorate each method with @abstractmethod; the body can simply be `pass`.
# ============================================================
class TransactionCommand(ABC):
    """Command Interface for financial operations."""

    # TODO: declare the two abstract methods listed in STEP 1
    pass

File: java-gdb-activities-20/activity17/src/com/gdb/command/TransactionCommand.java
package com.gdb.command;

import com.gdb.domain.Transaction;
import java.io.Serializable;

/**
 * Command interface defining common operations for financial transaction commands.
 */
public interface TransactionCommand extends Serializable {
    // ============================================================
    // 📝 STEP 1: Add execute() Method Signature
    //
    // INSTRUCTIONS:
    //   Declare void execute() throws Exception
    // ============================================================
    // TODO: declare execute() method
    void execute() throws Exception;
    
    // ============================================================
    // 📝 STEP 2: Add getTransaction() Method Signature
    //
    // INSTRUCTIONS:
    //   Declare Transaction getTransaction()
    // ============================================================
    // TODO: declare getTransaction() method
    Transaction getTransaction();
}
